disperser/controller/signature_receiver.go

```diff
@@ -569,3 +595,98 @@ func (sr *signatureReceiver) debugEquivalenceError(
 		)
 	}
 }
+
+// This method should be called when the controller is finished collecting signatures for a batch.
+func (sr *signatureReceiver) reportBatchMetrics() {
+	if sr.metrics == nil {
+		return
+	}
+
+	sr.reportThresholdSignedToDoneLatency()
+	sr.metrics.reportAttestationUpdateCount(float64(sr.attestationUpdateCount))
+
+	finalAttestation := sr.latestAttestation
+	if finalAttestation == nil {
+		sr.logger.Errorf("no final attestation to report metrics for batch %s",
+			hex.EncodeToString(sr.batchHeaderHash[:]))
+		return
+	}
+
+	batchHeaderHashString := hex.EncodeToString(sr.batchHeaderHash[:])
+
+	// Update global signing metrics.
+	for quorumID := range sr.indexedOperatorState.Operators {
+		quorumResults, ok := finalAttestation.QuorumResults[quorumID]
+		if !ok {
+			// Some unit tests trigger this
+			sr.logger.Errorf("missing quorum results for quorum %d in final attestation for batch %s",
+				quorumID, batchHeaderHashString)
+			continue
+		}
+
+		signingFraction := float64(quorumResults.PercentSigned) / 100.0
+		sr.metrics.ReportGlobalSigningThreshold(
+			quorumID,
+			sr.batchSizeBytes,
+			signingFraction)
+	}
+
+	// Update per-validator metrics
+	for quorumID, validatorsInQuorum := range sr.indexedOperatorState.Operators {
+
+		quorumTotals, ok := sr.indexedOperatorState.Totals[quorumID]
+		if !ok {
+			sr.logger.Errorf("missing quorum totals for quorum %d in final attestation for batch %s",
+				quorumID, batchHeaderHashString)
+			continue
+		}
+		quorumStake := quorumTotals.Stake
+
+		for operatorID := range validatorsInQuorum {
+			_, signed := sr.validSignerMap[operatorID]
+
+			quorumInfo, ok := sr.indexedOperatorState.Operators[quorumID]
```

Comment:
I'm confused here... you're getting `sr.indexedOperatorState.Operators[quorumID]`, and assigning the value to `quorumInfo`

But isn't the value of `sr.indexedOperatorState.Operators[quorumID]` already stored in `validatorsInQuorum`?